deepchem/models/normalizing_flows.py

```diff
@@ -0,0 +1,297 @@
+  complex distribution that is learned from data. The base distribution 
+  p(x) is transformed by the associated normalizing flow y=g(x) to model the
+  distribution p(y).
+
+  Normalizing flows combine the advantages of autoregressive models
+  (which provide likelihood estimation but do not learn features) and
+  variational autoencoders (which learn feature representations but
+  do not provide marginal likelihoods).
+
+  """
+
+  def __init__(self, base_distribution, flow_layers, **kwargs):
+    """Create a new NormalizingFlow.
+
+    Parameters
+    ----------
+    base_distribution : tfd.Distribution
+      Probability distribution to be transformed.
+      Typically an N dimensional multivariate Gaussian.
+    flow_layers : Sequence[tfb.Bijector]
+      An iterable of bijectors that comprise the flow.
+
+    """
+
+    try:
+      import tensorflow_probability as tfp
+      tfd = tfp.distributions
+      tfb = tfp.bijectors
+    except ModuleNotFoundError:
+      raise ValueError(
+          "This class requires tensorflow-probability to be installed.")
+
+    self.base_distribution = base_distribution
+    self.flow_layers = flow_layers
+
+    # Chain of flows is also a normalizing flow
+    bijector = tfb.Chain(list(reversed(self.flow_layers)))
+
+    # An instance of tfd.TransformedDistribution
+    self.flow = tfd.TransformedDistribution(
+        distribution=self.base_distribution, bijector=bijector)
+
+    super(NormalizingFlow, self).__init__(**kwargs)
+
+  def __call__(self, *inputs, training=True):
+    return self.flow.bijector.forward(*inputs)
+
+
+class NormalizingFlowModel(KerasModel):
+  """A base distribution and normalizing flow for applying transformations.
+
+  A distribution implements two main operations:
+    1. Sampling from the transformed distribution.
+    2. Calculating log probabilities.
+
+  A normalizing flow implements three main operations:
+    1. Forward transformation, 2. Inverse transformation, and 
+    3. Calculating the Jacobian.
+
+  Deep Normalizing Flow models require normalizing flow layers where
+  input and output dimensions are the same, the transformation is invertible,
+  and the determinant of the Jacobian is efficient to compute and
+  differentiable. The determinant of the Jacobian of the transformation 
+  gives a factor that preserves the probability volume to 1 when transforming
+  between probability densities of different random variables.
+
+  They are effective for any application requiring a probabilistic
+  model with these capabilities, e.g. generative modeling,
+  unsupervised learning, or probabilistic inference. For a thorough review
+  of normalizing flows, see [1]_.
+
+  References
+  ----------
+  .. [1] Papamakarios, George et al. "Normalizing Flows for Probabilistic Modeling and Inference." (2019). https://arxiv.org/abs/1912.02762.
+
+  """
+
+  def __init__(self, model: NormalizingFlow, **kwargs):
+    """Creates a new NormalizingFlowModel.
+
+    Parameters
```

Comment:
Could you add one sentence above this?  "In addition to the following arguments, this class also accepts all the keyword arguments from KerasModel."